pyomo/core/base/logical_constraint.py

```diff
@@ -495,61 +469,24 @@ def construct(self, data=None):
             return
         self._constructed = True
 
-        generate_debug_messages = is_debug_set(logger)
-        if generate_debug_messages:
-            logger.debug("Constructing logical constraint list %s" % self.name)
+        if is_debug_set(logger):
+            logger.debug("Constructing logical constraint list %s" % (self.name))
 
         if self._anonymous_sets is not None:
             for _set in self._anonymous_sets:
                 _set.construct()
 
-        assert self._init_expr is None
-        _init_rule = self.rule
-
-        #
-        # We no longer need these
-        #
-        self._init_expr = None
-        # Utilities like DAE assume this stays around
-        # self.rule = None
-
-        if _init_rule is None:
-            return
-
-        _generator = None
-        _self_parent = self._parent()
-        if inspect.isgeneratorfunction(_init_rule):
-            _generator = _init_rule(_self_parent)
-        elif inspect.isgenerator(_init_rule):
-            _generator = _init_rule
-        if _generator is None:
-            while True:
-                val = len(self._index_set) + 1
-                if generate_debug_messages:
-                    logger.debug("   Constructing logical constraint index " + str(val))
-                expr = apply_indexed_rule(self, _init_rule, _self_parent, val)
-                if expr is None:
-                    raise ValueError(
-                        "LogicalConstraintList '%s': rule returned None "
-                        "instead of LogicalConstraintList.End" % (self.name,)
-                    )
-                if (expr.__class__ is tuple) and (expr == LogicalConstraintList.End):
-                    return
-                self.add(expr)
-
-        else:
-            for expr in _generator:
-                if expr is None:
-                    raise ValueError(
-                        "LogicalConstraintList '%s': generator returned None "
-                        "instead of LogicalConstraintList.End" % (self.name,)
-                    )
-                if (expr.__class__ is tuple) and (expr == LogicalConstraintList.End):
-                    return
-                self.add(expr)
+        if self.rule is not None:
+            _rule = self.rule(self.parent_block(), ())
+            for cc in iter(_rule):
```

Comment:
What does `cc` stand for?

Reply:
Not entirely sure - it is just a temporary, and was copied from the implementation in ConstraintList.